### PR TITLE
pre-commit code document

### PR DESCRIPTION
**Pull request by YuvHayun**
## Related Issues
fixes: [CIAC-10010](https://jira-dc.paloaltonetworks.com/browse/CIAC-10010)

### CODE REVIEW

File: demisto_sdk/commands/pre_commit/README.md [hunk 1]
```diff
@@ -18,6 +18,65 @@ It utilizes the [pre-commit](https://github.com/pre-commit/pre-commit) infrastru
...
+* **--commited-only**
+Whether to run on committed files only.
+* **-g, --git-diff**
+Whether to use git to determine which files to run on.
+* **--prev-ver**
+The previous version to compare against. If not provided, the previous version will be determined using git.
+* **-a, --all-files**
+Whether to run on all files.
+* **--mode**
+Special mode to run the pre-commit with.
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/pre_commit/README.md:35]: this isnt clear enough IMO. we need docs for how to use properly (and this should give a link)

File: demisto_sdk/commands/pre_commit/README.md [hunk 2]
```diff
@@ -18,6 +18,65 @@ It utilizes the [pre-commit](https://github.com/pre-commit/pre-commit) infrastru
...
+* **--skip**
+A list of precommit hooks to skip.
+* **--no-validate**
+Whether to run without demisto-sdk validate.
+* **--format/--no-format**
+Whether to run demisto-sdk format.
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/pre_commit/README.md:41]: this flags works?

**Review by JudahSchwartz — APPROVED** [review]

File: demisto_sdk/commands/pre_commit/README.md (hunk 2 above)
Comment (YuvHayun) [reply demisto_sdk/commands/pre_commit/README.md:41]: yap, tested them myself.

File: demisto_sdk/commands/pre_commit/README.md (hunk 1 above)
Comment (YuvHayun) [reply demisto_sdk/commands/pre_commit/README.md:35]: There's a whole section for modes

**Pull request merged**